packages/block-editor/src/components/link-control/search-input.js

```diff
@@ -46,13 +46,16 @@ const LinkControlSearchInput = ( {
 		}
```

Comment:
It wasn't implemented here, but it's especially noticeable in reusing this for paragraph links: `selectItemHandler`'s treatment of `suggestion` doesn't really conform to the intended semantics of `URLInput`.

In this implementation: If a `suggestion` is provided as an argument to `selectItemHandler`, it's assumed to be the intended final result to be applied as the link, that the popover should be closed, and the transaction be considered "completed".

In the previous implementation: The closing of the popover is separate from the link selection.

Consider tabbing (related to https://github.com/WordPress/gutenberg/pull/19462#issuecomment-573828833). In the `master` branch, if I press <kbd>Tab</kbd> after having used arrow keys to find a suggestion, the link is "selected", but the tab proceeds to navigate through items in the popover, not close the popover.

![tabbing](https://user-images.githubusercontent.com/1779930/72286686-a2b0dd80-3613-11ea-8750-cdce8d5e87d1.gif)

But in this branch: When I press <kbd>Tab</kbd> after having selected a suggestion using arrow keys, the popover will close prematurely. That's because [the handling of <kbd>Tab</kbd>](https://github.com/WordPress/gutenberg/blob/4aad528337cff1bf7a20168fb442de613c9ea1e0/packages/block-editor/src/components/url-input/index.js#L208-L215) considers it as having made a selection.

![tabbing-2](https://user-images.githubusercontent.com/1779930/72286889-0c30ec00-3614-11ea-8c7d-47e9b8682932.gif)


I guess this is why the `<form onSubmit>` handling was implemented separate from `URLInput`, since we want the <kbd>Enter</kbd> press to be considered as both a selection (via `URLInput`) and as a completion of the transaction (from this component).